overlays/nodejs/cloudwatch/logGroupMixins.ts

```diff
@@ -0,0 +1,148 @@
+    // records with a "CONTROL_MESSAGE" type, mainly for checking if the destination is reachable.
+    messageType: string;
+
+    // The actual log data, represented as an array of log event records. The "id" property is a
+    // unique identifier for every log event.
+    logEvents: LogGroupEventRecord[];
+}
+
+export interface LogGroupEventRecord {
+    id: string;
+    timestamp: number;
+    message: string;
+}
+
+export type LogGroupEventHandler = lambda.EventHandler<LogGroupEvent, void>;
+
+export class LogGroupEventSubscription extends lambda.EventSubscription {
+    public readonly logGroup: pulumi.Output<logGroup.LogGroup>;
+    public readonly logSubscriptionFilter: logSubscriptionFilter.LogSubscriptionFilter;
+
+    constructor(
+        name: string, logGroup: logGroup.LogGroup, handler: LogGroupEventHandler,
+        args: LogGroupEventSubscriptionArgs, opts?: pulumi.ResourceOptions) {
+
+        super("aws:cloudwatch:LogGroupEventSubscription", name, { logGroup: logGroup }, opts);
+
+        args = args || {};
+        const parentOpts = { parent: this };
+        this.func = lambda.createFunctionFromEventHandler(name, handler, parentOpts);
+
+        this.permission = new lambda.Permission(name, {
+            action: "lambda:invokeFunction",
+            function: this.func,
+            principal: `logs.${config.requireRegion()}.amazonaws.com`,
+            sourceArn: logGroup.arn,
+        }, parentOpts);
+
+        this.logSubscriptionFilter = new logSubscriptionFilter.LogSubscriptionFilter(name, {
+            destinationArn: this.func.arn,
+            filterPattern: args.filterPattern || "",
+            logGroup: logGroup,
+        }, parentOpts);
+    }
+}
+
+declare module "./logGroup" {
+    interface LogGroup {
+        /**
+         * Creates a new subscription to events fired from this LogGroup to the handler provided,
+         * along with options to control the behavior of the subscription.
+         *
+         * The events will be produced in raw (gzipped + base64 encoded) form.
+         */
+        onEvent(name: string, handler: LogGroupEventHandler,
+                args?: LogGroupEventSubscriptionArgs, opts?: pulumi.ResourceOptions): LogGroupEventSubscription;
+
+        /**
+         * Creates a new subscription to events fired from this LogGroup to the handler provided,
+         * along with options to control the behavior of the subscription.
+         *
+         * The events will be provided in their decoded form.  Because this event hookup needs to
+         * execute code to convert the raw messages, it can only be passed an [EntryPoint] callback,
+         * not a [lambda.Function] instance.
+         */
+        onDecodedEvent(name: string, handler: lambda.EntryPoint<DecodedLogGroupEvent, void>,
+            args?: LogGroupEventSubscriptionArgs, opts?: pulumi.ResourceOptions): LogGroupEventSubscription;
```

Comment:
Note: this is an example of how we can make things much nicer for users.  Working with a loggroup event kinda sucks (it's zipped and base64 compressed).  But we can give a nice entrypoint that unwraps that for you and allows you easy access to the original data.